modules/consensus/validtransaction.go

```diff
@@ -208,11 +208,11 @@ func (s *State) validTransaction(t types.Transaction) error {
 	if err != nil {
 		return err
 	}
-	err = s.validFileContractRevisions(t)
+	err = s.validStorageProofs(t)
 	if err != nil {
 		return err
 	}
-	err = s.validStorageProofs(t)
+	err = s.validFileContractRevisions(t)
```

Comment:
I'm not 100% sure why I chose that order, but a lot of the verification steps need to be done in an explicit order.

In this case, I **think** it's okay, but we should probably leave it stable unless there's a pressing reason to change it.